src/libstd/keyword_docs.rs

```diff
@@ -116,14 +126,21 @@ mod const_keyword { }
 /// [Reference]: ../reference/items/extern-crates.html
 mod crate_keyword { }
 
+#[doc(keyword = "else")]
+//
+/// What to do when an [`if`] condition is false.
+///
+/// [`if`]: keyword.if.html
+mod else_keyword { }
+
 #[doc(keyword = "enum")]
 //
-/// For defining enumerations.
+/// An [Algebraic Data Type][ADT].
```

Comment:
A `struct` is also an ADT; `enum` is specifically a *sum type*.

Reply:
True. This needs to get changed. FWIW, I think it would be better to use  "composite" and "enumerated" in the summary line as opposed to "product" and "sum", since the audience for these docs is unlikely to be familiar with them (although "composite" and "enumerated" aren't much clearer).

Reply:
I think "composite" isn't that familiar either. At least "product" is a bit familiar for those who have heard of a Cartesian product. However, since the goal here is to give an intuitive explanation, we can avoid both terminologies altogether:
```suggestion
/// A type that can be one of several variants.
```
or alternatively:
```suggestion
/// A type encoding the notion of choice.
```